dataline-scheduler/src/test/java/io/dataline/scheduler/DefaultSchedulerPersistenceTest.java

```diff
@@ -0,0 +1,338 @@
+    destinationConnectionImplementation.setConfiguration(implementationJson);
+
+    final Column column = new Column();
+    column.setDataType(DataType.STRING);
+    column.setName("id");
+    column.setSelected(true);
+
+    final Table table = new Table();
+    table.setName("users");
+    table.setColumns(Lists.newArrayList(column));
+    table.setSelected(true);
+
+    final Schema schema = new Schema();
+    schema.setTables(Lists.newArrayList(table));
+
+    final UUID connectionId = UUID.randomUUID();
+
+    standardSync = new StandardSync();
+    standardSync.setConnectionId(connectionId);
+    standardSync.setName("presto to hudi");
+    standardSync.setStatus(StandardSync.Status.ACTIVE);
+    standardSync.setSchema(schema);
+    standardSync.setSourceImplementationId(sourceImplementationId);
+    standardSync.setDestinationImplementationId(UUID.randomUUID());
+    standardSync.setSyncMode(StandardSync.SyncMode.APPEND);
+
+    now = Instant.now();
+    @SuppressWarnings("unchecked")
+    Supplier<Instant> timeSupplier = mock(Supplier.class);
+    when(timeSupplier.get()).thenReturn(now);
+
+  }
+
+  @SuppressWarnings("rawtypes")
+  @BeforeAll
+  public static void dbSetup() {
+    container = new PostgreSQLContainer("postgres:13-alpine")
+        .withDatabaseName("dataline")
+        .withUsername("docker")
+        .withPassword("docker");
+    container.start();
+
+    try {
+      container.copyFileToContainer(MountableFile.forClasspathResource("schema.sql"), "/etc/init.sql");
+      // execInContainer uses Docker's EXEC so it needs to be split up like this
+      container.execInContainer(
+          "psql", "-d", "dataline", "-U", "docker", "-a", "-f", "/etc/init.sql");
+
+    } catch (InterruptedException | IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    connectionPool =
+        DatabaseHelper.getConnectionPool(
+            container.getUsername(), container.getPassword(), container.getJdbcUrl());
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    container.stop();
+  }
+
+  @BeforeEach
+  public void setup() throws SQLException {
+    // todo (cgardens) - truncate whole db.
+    DatabaseHelper.query(connectionPool, ctx -> ctx.truncate("jobs"));
+
+  }
+
+  private Record getJobRecord(long jobId) throws SQLException {
+    return DatabaseHelper.query(
+        connectionPool,
+        ctx -> ctx.fetch("SELECT * FROM jobs WHERE id = ?", jobId).stream()
+            .findFirst()
+            .orElseThrow(
+                () -> new RuntimeException("Could not find job with id: " + jobId)));
+  }
+
+  @Test
+  public void testCreateSourceCheckConnectionJob() throws IOException, SQLException {
+    final SchedulerPersistence schedulerPersistence = new DefaultSchedulerPersistence(connectionPool);
+    final long jobId = schedulerPersistence.createSourceCheckConnectionJob(sourceConnectionImplementation);
+
+    Record jobEntry = getJobRecord(jobId);
```

Comment:
everything else is final, why not this one?

Reply:
bah. no good reason. i need to figure out how to lint this.